### DISCUSSION TITLE
Error when setting babel in my monorepo

### DISCUSSION THREAD

**Discussion by jotaachm** [babel/babel#16646] [Q&A]
Hey!
I am combining my React web application with my Expo React Native mobile app into a monorepo using yarn workspaces because they both need to share some utils, queries, etc. My file structure is something like this:
```
- ui
    - apps
        - mobile
            - package.json
            - babel.config.js
        - web
            - package.json
            - babel.config.json
    - packages
        - shared
            - package.json
    - package.json
    - node_modules
```

I am getting this error when trying to run the web app:
```
Module build failed (from ../../node_modules/babel-loader/lib/index.js):
SyntaxError: ui/node_modules/expo-router/build/ExpoRoot.js: Support for the experimental syntax 'jsx' isn't currently enabled (54:17):

  52 |      */
  53 |     const wrapper = ({ children }) => {
> 54 |         return (<ParentWrapper>
     |                 ^
  55 |         <react_native_safe_area_context_1.SafeAreaProvider 
  56 |         // SSR support
  57 |         initialMetrics={INITIAL_METRICS}>

Add @babel/preset-react (https://github.com/babel/babel/tree/main/packages/babel-preset-react) to the 'presets' section of your Babel config to enable transformation.
If you want to leave it as-is, add @babel/plugin-syntax-jsx (https://github.com/babel/babel/tree/main/packages/babel-plugin-syntax-jsx) to the 'plugins' section to enable parsing.

If you already added the plugin for this syntax to your config, it's possible that your config isn't being loaded.
You can re-run Babel with the BABEL_SHOW_CONFIG_FOR environment variable to show the loaded configuration:
    npx cross-env BABEL_SHOW_CONFIG_FOR=ui/node_modules/expo-router/build/ExpoRoot.js <your build command>
See https://babeljs.io/docs/configuration#print-effective-configs for more info.
```

I have babel.config.json file that looks like this in ui/apps/web
```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "edge": "17",
          "firefox": "60",
          "chrome": "67",
          "safari": "11.1"
        },
        "useBuiltIns": "usage",
        "corejs": "3.6.5"
      }
    ],
    "@babel/preset-react"
  ]
}
```

And a babel.config.json in ui/apps/mobile that looks like this:
```
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: ['nativewind/babel'],
  };
};
```

Reading [this](https://babeljs.io/docs/config-files), it seems that I should have a babel.config.json in my root directory for the monorepo. So I've tried moving the web babel.config.json to the root and reinstalling everything, but it still doesn't work. Am I doing something wrong? What's the ideal babel configuration for my project?

Any help is appreciated! Thanks in advance

**Comment by nicolo-ribaudo** [comment]
That's right, there can be only one `babel.config.json` file controlling how your code is compiled. Do you still get the same error when using a single one? How are you running Babel? From where?

**Reply by jotaachm** [reply]
Thanks for answering!
Yes, if I put a babel.config.json file at the root of my monorepo I still have the same error. I am using Create React App, so I think I am not running babel because CRA handles it.
